onnxruntime/core/session/inference_session.cc

```diff
@@ -517,7 +517,13 @@ common::Status InferenceSession::RegisterExecutionProvider(const std::shared_ptr
       ORT_RETURN_IF_ERROR(p_exec_provider->SetComputeStream(trt_ep->GetComputeStream()));
     }
   }
-
+#ifndef DNNL_OPENMP
+  // This is safetly called only when we are not using OpenMP, preprocessor directive will probably be
+  // removed in the future
+  if (provider_type == onnxruntime::kDnnlExecutionProvider) {
+    ORT_RETURN_IF_ERROR(p_exec_provider->SetComputeStream(static_cast<void*>(GetIntraOpThreadPoolToUse())));
```

Comment:
Not sure I'm following this. I suppose you're reusing the void* of the compute stream for the ORT thread pool. This is super confusing and compromises readability.

Reply:
We need to think this through. I don't think the changes in this PR to achieve this in the current form are acceptable.

Reply:
@pranavsharma we are open to suggestions as well as testing out alternate proposals.  Do you have any suggestions?